### PR TITLE
TECH-432: Address State Integration Testing

### PR DESCRIPTION
**Pull request by parousisv**
This is the task regarding the integration testing on the changes made on PRs #66 and #74.

### CODE REVIEW

**Review by c4t-ag — APPROVED** [review]
LGTM!

**Review by c4t-ag — CHANGES_REQUESTED** [review]
Please PR against dev branch, thx!

**Review by c4t-ag — APPROVED** [review]

**Review by c4t-ag — CHANGES_REQUESTED** [review]
There are foreign commits in the PR

**Review by charalarg — CHANGES_REQUESTED** [review]
IMO I think we also need to cover with tests these two cases too, maybe with different combinations:
- https://github.com/chain4travel/caminogo/blob/TECH-432-Address_State_Integration_Testing/vms/platformvm/txs/executor/camino_tx_executor.go#L532
- https://github.com/chain4travel/caminogo/blob/TECH-432-Address_State_Integration_Testing/vms/platformvm/txs/executor/camino_tx_executor.go#L536

Also, there are no tests triggering errors coming from `VerifySpend` in the executor and from `Lock` function in the builder, but i don't think we need them here cause these function are tested with unit tests so i guess we assume that they work as expected. But i need your opinion here too @evlekht .

**Comment by evlekht** [comment]
> Also, there are no tests triggering errors coming from VerifySpend in the executor and from Lock function in the builder, but i don't think we need them here cause these function are tested with unit tests so i guess we assume that they work as expected. But i need your opinion here too @evlekht .

regarding this, I think its ok to leave VerifySpend fail cases untested, thats more general logic and I don't expect that will duplicate this tests for every tx

**Comment by evlekht** [comment]
does it covers the case, when tx sigs doesn't contain addr with state that authorizes to set other addresses state ?

File: vms/platformvm/txs/executor/camino_tx_executor_test.go [hunk 1]
```diff
@@ -547,3 +547,266 @@ func TestCaminoAddValidatorTxNodeSig(t *testing.T) {
...
+			stateAddress:  ids.ShortEmpty,
+			targetAddress: ids.ShortEmpty,
+			flag:          uint64(txs.AddressStateRoleKyc),
+			state:         txs.AddressStateRoleKyc,
+			expectedErr:   txs.ErrEmptyAddress,
+			remove:        false,
+		},
+	}
+
+	generateExecutor := func(unsidngedTx txs.UnsignedTx, env *environment) CaminoStandardTxExecutor {
```
Comment (evlekht) [inline vms/platformvm/txs/executor/camino_tx_executor_test.go:767]: there is no point in making this func, imo
u'll still use it in only one place

Comment (parousisv) [reply vms/platformvm/txs/executor/camino_tx_executor_test.go]: OK, removed

**Review by charalarg — APPROVED** [review]

**Pull request merged**